packages/dropdowns/src/Menu/Menu.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+import { Popper } from 'react-popper';
+import { withTheme, isRtl } from '@zendeskgarden/react-theming';
+
+import { StyledMenu } from '../styled/index.js';
+import useDropdownContext from '../utils/useDropdownContext';
+import { getPopperPlacement, getRtlPopperPlacement } from '../utils/garden-placements';
+
+export const MenuContext = createContext();
+
+/**
+ * Accepts all `<ul>` props
+ */
+const Menu = props => {
+  const {
+    placement,
+    popperModifiers,
+    eventsEnabled,
+    animate,
+    maxHeight,
+    style: menuStyle,
+    zIndex,
+    ...otherProps
+  } = props;
+  const {
+    itemIndexRef,
+    previousItemRef,
+    previousIndexRef,
+    nextItemsHashRef,
+    popperReferenceElementRef,
+    downshift: { isOpen, getMenuProps }
+  } = useDropdownContext();
+
+  // Reset Downshift refs on every render
+  itemIndexRef.current = 0;
+  nextItemsHashRef.current = {};
+  previousItemRef.current = undefined;
+  previousIndexRef.current = undefined;
+
+  const popperPlacement = isRtl(props)
+    ? getRtlPopperPlacement(placement)
+    : getPopperPlacement(placement);
+
+  return (
+    <MenuContext.Provider value={{ itemIndexRef }}>
+      <Popper placement={popperPlacement} modifiers={popperModifiers} eventsEnabled={eventsEnabled}>
+        {({ ref, style, scheduleUpdate, placement: currentPlacement }) => {
+          let computedStyle = menuStyle;
+
+          // Calculate custom width if ref is provided from Select or Autocomplete
+          if (popperReferenceElementRef.current) {
+            computedStyle = {
+              ...menuStyle,
+              width: popperReferenceElementRef.current.getBoundingClientRect().width
```

Comment:
Should we check if `getBoundingClientRect` is available this will break SSR right?

Reply:
I guess it would depend on the SSR environment. I've added another check for `getBoundingClientRect`